src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -971,31 +954,57 @@ private void AddEasyHandle(RequestCompletionSource state)
             }
         }
 
+        private static void RemoveEasyHandle(SafeCurlMultiHandle multiHandle, GCHandle stateHandle, RequestCompletionSource state)
+        {
+            RemoveEasyHandle(multiHandle, state, false);
```

Comment:
Can this RemoveEasyHandle call be moved down to below the if block?  Ideally we dispose of these resources as soon as we no longer need them, so that the GC doesn't have to track it for longer and for a bit more reliability in call to RemoveEasyHandle were to throw.